sentry/src/event_aggregator.rs

```diff
@@ -138,11 +139,25 @@ impl EventAggregator {
             AccessError::OnlyCreatorCanCloseChannel | AccessError::ForbiddenReferrer => {
                 ResponseError::Forbidden(e.to_string())
             }
+            AccessError::OnlyCreatorCanUpdateTargetingRules => {
+                ResponseError::Forbidden(e.to_string())
+            }
             AccessError::RulesError(error) => ResponseError::TooManyRequests(error),
             AccessError::UnAuthenticated => ResponseError::Unauthorized,
             _ => ResponseError::BadRequest(e.to_string()),
         })?;
 
+        let events_stream = stream::iter(events);
+        let events_eval = events_stream.then(|ev| async move {
+            match ev {
+                Event::UpdateTargeting { targeting_rules } => {
+                    update_targeting_rules(&app.pool, &channel_id, &targeting_rules).await
+                }
+                _ => Ok(false),
+            }
+        });
+        events_eval.collect::<Vec<_>>().await;
```

Comment:
cool! I didn't know about this trick with `Stream`.

does `collect()` creates a `Vec<Result<bool, ...>>` in this case? Maybe we can collect in single `Result` and check if it's `Ok(true)` otherwise we can return a `400 Bad Request` maybe?

Reply:
Yes, it is `Vec<Result<bool, ...>>`, if it is`Ok(false)` it means it's just not an `UpdateTargeting` event.